### PR TITLE
Make OFFXML serialization more backwards-compatible

### LINKED ISSUE #1278: New toolkit writes XML files that cannot be parsed by old versions
There is currently custom logic for writing unit-bearing quantites to OFFXML files: https://github.com/openforcefield/openff-toolkit/blob/0.10.5/openff/toolkit/utils/utils.py#L207-L233

I removed this logic when moving to `openff-units` (https://github.com/openforcefield/openff-toolkit/pull/1142): https://github.com/openforcefield/openff-toolkit/blob/3f13de16094ee92cd9f70a835e103a6e90d10efc/openff/toolkit/utils/utils.py#L190-L216

The resulting OFFXMLs can be read by "new" toolkit versions but not "old" versions. This backwards compatibility is nice to have and should be fixed by adding the custom logic back.

### PR DESCRIPTION
**Pull request by mattwthompson**
Fixes #1278 

This can't easily be tested (I'm not interested in a CI job that switches out multiple conda environments, including swapping to an old one) but it works with a quick pass:

```
(openff-dev) [openff-toolkit] git branch | cat                                     11:10:08  ☁  offxml-roundtrip ☂
  master
* offxml-roundtrip
  prep-off-ep-0005b
  version-object
  version-object2
(openff-dev) [openff-toolkit] python                                               11:10:11  ☁  offxml-roundtrip ☂
Python 3.9.12 | packaged by conda-forge | (main, Mar 24 2022, 23:23:20)
[Clang 12.0.1 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from openff.toolkit.typing.engines.smirnoff import ForceField
>>> sage = ForceField("openff-2.0.0.offxml")
>>> sage.to_file("sage.offxml")
>>>
(openff-dev) [openff-toolkit] git checkout 0.10.5                                  11:10:36  ☁  offxml-roundtrip ☂
Note: switching to '0.10.5'.
HEAD is now at cba4ae38 Refactor virtual site handler (#1252)
(openff-dev) [openff-toolkit] conda deactivate                                             11:10:40  ☁  cba4ae38 ☂
(base) [openff-toolkit] conda activate openff-old                                          11:10:44  ☁  cba4ae38 ☂
(openff-old) [openff-toolkit] python                                                       11:10:47  ☁  cba4ae38 ☂
Python 3.9.12 | packaged by conda-forge | (main, Mar 24 2022, 23:23:20)
[Clang 12.0.1 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from openff.toolkit.typing.engines.smirnoff import ForceField
>>> ForceField("sage.offxml")
<openff.toolkit.typing.engines.smirnoff.forcefield.ForceField object at 0x10d6fe070>
```

at least in terms of the file being read.

I have to add that I think this is a gnarly approach and opens the door to compromises elsewhere. It would be nice to quarantine the code responsible for this behavior, as I have loosely done in this PR, and commit to removing it and the corresponding behavior at a certain point in time. Bending over backwards to ensure the string representation between OpenFF units and OpenMM are so precise is not something I think we should be committed to indefinitely, whereas round-tripping files within the same release of the toolkit is something we should be concerned with.

- [x] Tag issue being addressed
- [ ] Add [tests](https://github.com/openforcefield/openff-toolkit/tree/master/openff/toolkit/tests)
- [ ] Update docstrings/[documentation](https://github.com/openforcefield/openff-toolkit/tree/master/docs), if applicable
- [x] [Lint](https://open-forcefield-toolkit.readthedocs.io/en/latest/developing.html#style-guide) codebase
- [ ] Update [changelog](https://github.com/openforcefield/openff-toolkit/blob/master/docs/releasehistory.rst)

### CODE REVIEW

File: openff/toolkit/utils/utils.py [hunk 1]
```diff
@@ -164,8 +164,33 @@ def get_data_file_path(relative_path):
     return fn
 
 
-def unit_to_string(input_unit):
-    return str(input_unit)
```
Comment (mattwthompson) [inline openff/toolkit/utils/utils.py:168]: I found this was never hit in tests, and I'm not quite sure how it would be triggered.

**Comment by mattwthompson** [comment]
A simpler approach is to override setting Pint's default formatting to a compact, pretty one: https://github.com/openforcefield/openff-units/blob/bdb6483ee9bfd03051fcd88c478623926c464dd7/openff/units/units.py#L71


```
(openff-dev) [openff-toolkit] git branch | cat                                                         9:59:37  ☁  offxml-roundtrip ☂ ✭
  flake8-0-11-x
  master
* offxml-roundtrip
  prep-off-ep-0005b
  registry-manager
  version-object
  version-object2
(openff-dev) [openff-toolkit] python                                                                   9:59:38  ☁  offxml-roundtrip ☂ ✭
Python 3.9.12 | packaged by conda-forge | (main, Mar 24 2022, 23:23:20)
[Clang 12.0.1 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from openff.toolkit.typing.engines.smirnoff import ForceField
>>> sage = ForceField("openff-2.0.0.offxml")
>>> sage.to_file("test_roundtrip.offxml")
>>>
(openff-dev) [openff-toolkit] conda deactivate && conda activate openff-old && git checkout 0.10.5    10:00:09  ☁  offxml-roundtrip ☂ ✭
HEAD is now at cba4ae38 Refactor virtual site handler (#1252)
(openff-old) [openff-toolkit] python                                                                          10:00:14  ☁  cba4ae38 ☂ ✭
Python 3.9.12 | packaged by conda-forge | (main, Mar 24 2022, 23:23:20)
[Clang 12.0.1 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from openff.toolkit.typing.engines.smirnoff import ForceField
/Users/mattthompson/miniconda3/envs/openff-old/lib/python3.9/site-packages/setuptools/version.py:1: UserWarning: Module openff was already imported from None, but /Users/mattthompson/software/openff-toolkit is being added to sys.path
  import pkg_resources
Warning: Unable to load toolkit 'OpenEye Toolkit'. The Open Force Field Toolkit does not require the OpenEye Toolkits, and can use RDKit/AmberTools instead. However, if you have a valid license for the OpenEye Toolkits, consider installing them for faster performance and additional file format support: https://docs.eyesopen.com/toolkits/python/quickstart-python/linuxosx.html OpenEye offers free Toolkit licenses for academics: https://www.eyesopen.com/academic-licensing
>>> ForceField("test_roundtrip.offxml")
<openff.toolkit.typing.engines.smirnoff.forcefield.ForceField object at 0x10a0790d0>
```

this may not cover all cases and there may be differences in how things look on disk, but if there are they seem to be within the set of things OpenMM seems to infer the correct meaning of.

**Comment by mattwthompson** [comment]
FWIW I re-ran the manual check from the last comment and it's still loading without obvious error.

**Review by j-wags — APPROVED** [review]
Tested locally and this works great. Thanks @mattwthompson

**Pull request merged**